pkg/compiler/compiler.go

```diff
@@ -46,15 +58,16 @@ func (c *Compiler) CompileTests() error {
 		return microerror.Mask(err)
 	}
 
-	name := harness.GetProjectName()
+	e2eBinary := harness.GetProjectName() + "-e2e"
+	e2eDir := filepath.Join(dir, c.testDir)
```

Comment:
Test dir can now be `integration`, `integration/upgrade`, `2e2/foo`, or whatever users like.